lib/matplotlib/tests/test_artist.py

```diff
@@ -372,3 +374,117 @@ class MyArtist4(MyArtist3):
         pass
 
     assert MyArtist4.set is MyArtist3.set
+
+
+def test_format_cursor_data_BoundaryNorm():
+    """Test if cursor data is correct when using BoundaryNorm."""
+    X = np.empty((3, 3))
+    X[0, 0] = 0.9
+    X[0, 1] = 0.99
+    X[0, 2] = 0.999
+    X[1, 0] = -1
+    X[1, 1] = 0
+    X[1, 2] = 1
+    X[2, 0] = 0.09
+    X[2, 1] = 0.009
+    X[2, 2] = 0.0009
+
+    # map range -1..1 to 0..256 in 0.1 steps
+    fig, ax = plt.subplots()
+    fig.suptitle("-1..1 to 0..256 in 0.1")
+    norm = mcolors.BoundaryNorm(np.linspace(-1, 1, 20), 256)
+    img = ax.imshow(X, cmap='RdBu_r', norm=norm)
+    for v in X.flat:
+        label = "[{:-#.{}g}]".format(v, cbook._g_sig_digits(v, 0.1))
```

Comment:
This test just tests that the code is called, not that the resulting labels are correct.  Please consider writing out explicitly what the labels should be and comparing those.

Reply:
> So I don't know what to do? Do you want me to just replace l. 398 with the verbatim output of the format call?

Yes, I think so.  In general it is bad for the tests to include private methods, though that is a rule meant to be broken...

Reply:
Agreed, it is more understandable that way. But I left the original lines with the `cbook` as comments.